module/API/Controller.cpp

```diff
@@ -1,68 +1,106 @@
 /**
  * @file Controller.cpp
  * @brief モーター制御に用いる関数をまとめたラッパークラス
- * @author takahashitom
+ * @author takahashitom CHIHAYATAKU
  */
 #include "Controller.h"
 
 Controller::Controller() : rightWheel(PORT_B), leftWheel(PORT_C), armMotor(PORT_A) {}
 
+// PWMの初期化
+double Controller::pwmOfRightWheel = 0.0;
+double Controller::pwmOfLeftWheel = 0.0;
+double Controller::pwmOfArm = 0.0;
+
 // モータに設定するPWM値の制限
-int Controller::limitPwmValue(const int value)
+double Controller::limitPwmValue(const double inputPwm)
 {
-  if(value > MOTOR_PWM_MAX) {
+  if(inputPwm > MOTOR_PWM_MAX) {
     return MOTOR_PWM_MAX;
-  } else if(value < MOTOR_PWM_MIN) {
+  } else if(inputPwm < MOTOR_PWM_MIN) {
     return MOTOR_PWM_MIN;
   }
-  return value;
+  return inputPwm;
 }
 
 // 右モータにPWM値をセット
-void Controller::setRightMotorPwm(const int pwm)
+void Controller::setRightMotorPwm(const double pwm)
 {
-  rightWheel.setPWM(limitPwmValue(pwm));
+  if(pwm > MOTOR_PWM_MAX) {
+    pwmOfRightWheel = MOTOR_PWM_MAX;
+  } else if(pwm < MOTOR_PWM_MIN) {
+    pwmOfRightWheel = MOTOR_PWM_MIN;
+  } else {
+    pwmOfRightWheel = pwm;
+  }
+  rightWheel.setPWM(int(limitPwmValue(pwm)));
```

Comment:
同じようなことをやってるlimitPwmValueがあるので、以下のように変更してほしいです。
それと、度々申し訳ないのだけど、static_castを使ったほうが、安全に型変換できるということに気づきました。
```suggestion
pwmOfRightWheel = limitPwmValue(pwm);
rightWheel.setPWM(static_cast<int>(pwmOfRightWheel));
```